app/assets/javascripts/qpixel_api.js

```diff
@@ -212,26 +213,16 @@ window.QPixel = {
     let prefs = await QPixel.preferences();
     let value = community ? prefs.community[name] : prefs.global[name];
 
-    // Deliberate === here: null is a valid value for a preference, but undefined means we haven't fetched it.
-    // If we haven't fetched a preference, that probably means it's new - run a full re-fetch.
-    if (value === undefined) {
-      const resp = await fetch('/users/me/preferences', {
-        credentials: 'include',
-        headers: {
-          'Accept': 'application/json'
-        }
-      });
-      const data = await resp.json();
-      localStorage['qpixel.user_preferences'] = JSON.stringify(data);
-      this._preferences = data;
-
-      prefs = await QPixel.preferences();
-      value = community ? prefs.community[name] : prefs.global[name];
-      return value;
-    }
-    else {
+    // Note that null is a valid value for a preference, but undefined means we haven't fetched it.
+    if (typeof(value) !== 'undefined') {
```

Comment:
In general I prefer to use `typeof` because it is more robust. Without it, direct comparison with `undefined` will result in a `ReferenceError` if `value` does not exist. For example:

Defining a variable as `undefined` causes no problem. The following outputs 'undefined' as expected:
```js
const value = undefined;
if (value !== undefined) {
    console.log('defined');
} else {
    console.log('undefined');
}
```

Declaring a variable but not defining it also causes no problem. The following also outputs 'undefined' as expected:
```js
let value;
if (value !== undefined) {
    console.log('defined');
} else {
    console.log('undefined');
}
```

Not declaring a variable causes a problem as it cannot be accessed in order to process the conditional expression of the `if` block. The following might be expected to output 'undefined', but it actually results in a `ReferenceError`:
```js
if (value !== undefined) {
    console.log('defined');
} else {
    console.log('undefined');
}
```

Using `typeof` avoids this problem, as it can cope with undeclared variables. The following outputs 'undefined':
```js
if (typeof(value) !== 'undefined') {
    console.log('defined');
} else {
    console.log('undefined');
}
```

The line you commented on follows a declaration of `value`, so in this case the robustness is not needed, but I wanted to make it robust to any future changes too.